pup.py

```diff
@@ -186,27 +180,21 @@ def make_producer(queue=None):
         os.remove(temp)
 
 
-async def extract_facts(archive):
-    logger.info("extracting facts from %s", archive)
-    facts = {}
-    try:
-        with extract(archive) as ex:
-            facts = get_canonical_facts(path=ex.tmp_dir)
-    except (InvalidArchive, ModuleNotFoundError, KeyError) as e:
-        facts['error'] = e.args[0]
-
-    return facts
-
-
 @time(mnm.inventory_post_time)
 async def post_to_inventory(facts, msg):
 
+    system_profile = facts.pop('system_profile')
     post = {**facts, **msg}
     post['account'] = post.pop('rh_account')
     post['facts'] = []
     if post.get('metadata'):
         post['facts'].append({'facts': post.pop('metadata'),
                               'namespace': 'insights-client'})
+        post['facts'].append({'facts': post.pop('system_profile'),
```

Comment:
I think we can remove this bit. We want it to drop the system_profile facts in there every time. the way it's written now, we'll be adding it twice. We can use the on below since it also references the `system_profile` variable rather than trying to pop it out of the dict